ogc/features/datasources/geopackage/geopackage.go

```diff
@@ -396,71 +364,24 @@ func (g *GeoPackage) getFeatureTable(collection string) (*featureTable, error) {
 	return table, nil
 }
 
-// Read gpkg_contents table. This table contains metadata about feature tables. The result is a mapping from
-// collection ID -> feature table metadata. We match each feature table to the collection ID by looking at the
-// 'identifier' column. Also in case there's no exact match between 'collection ID' and 'identifier' we use
-// the explicitly configured table name.
-func readGpkgContents(collections engine.GeoSpatialCollections, db *sqlx.DB) (map[string]*featureTable, error) {
-	query := `
-select
-	c.table_name, c.data_type, c.identifier, c.description, c.last_change,
-	c.min_x, c.min_y, c.max_x, c.max_y, c.srs_id, gc.column_name, gc.geometry_type_name
-from
-	gpkg_contents c join gpkg_geometry_columns gc on c.table_name == gc.table_name
-where
-	c.data_type = 'features' and 
-	c.min_x is not null`
-
-	rows, err := db.Queryx(query)
-	if err != nil {
-		return nil, fmt.Errorf("failed to retrieve gpkg_contents using query: %v\n, error: %w", query, err)
-	}
-	defer rows.Close()
-
-	result := make(map[string]*featureTable, 10)
-	for rows.Next() {
-		row := featureTable{}
-		if err = rows.StructScan(&row); err != nil {
-			return nil, fmt.Errorf("failed to read gpkg_contents record, error: %w", err)
-		}
-		if row.TableName == "" {
-			return nil, fmt.Errorf("feature table name is blank, error: %w", err)
-		}
-
-		if len(collections) == 0 {
-			result[row.Identifier] = &row
-		} else {
-			for _, collection := range collections {
-				if row.Identifier == collection.ID {
-					result[collection.ID] = &row
-					break
-				} else if hasMatchingTableName(collection, row) {
-					result[collection.ID] = &row
-					break
-				}
-			}
-		}
-	}
-
-	if err := rows.Err(); err != nil {
-		return nil, err
-	}
-	if len(result) == 0 {
-		return nil, fmt.Errorf("no records found in gpkg_contents, can't serve features")
-	}
-
-	return result, nil
-}
-
-func hasMatchingTableName(collection engine.GeoSpatialCollection, row featureTable) bool {
-	return collection.Features != nil && collection.Features.TableName != nil &&
-		row.Identifier == *collection.Features.TableName
-}
-
 func readGpkgGeometry(rawGeom []byte) (geom.Geometry, error) {
 	geometry, err := gpkg.DecodeGeometry(rawGeom)
 	if err != nil {
 		return nil, err
 	}
 	return geometry.Geometry, nil
 }
+
+func propertyFiltersToSQL(pf map[string]string) (sql string, namedParams map[string]any) {
+	namedParams = make(map[string]any)
+	if len(pf) > 0 {
+		position := 0
+		for k, v := range pf {
+			position++
+			namedParam := fmt.Sprintf("pf%d", position)
+			sql += fmt.Sprintf(" and %s = :%s", k, namedParam)
```

Comment:
`k` misschien quoten voor het geval de column name een keyword is

Reply:
goed punt